src/browser_utils.js

```diff
@@ -653,11 +653,29 @@ async function clickSelector(page, selector, {timeout=getDefaultTimeout(page), c
                 const element = document.querySelector(selector);
                 if (!element) return false;
 
-                if (visible && element.offsetParent === null) return null; // invisible
+                if (visible) {
+                    if (element.offsetParent === null) return null; // invisible
 
+                    const rect = /** @type {Element} */ (element).getBoundingClientRect();
+                    return {
+                        x: rect.x + (rect.width  / 2),
+                        y: rect.y + (rect.height / 2)
+                    };
+                }
+
+                // We can't use the mouse to click on invisible elements.
+                // Therefore invoke the click handler on the DOM node directly.
                 element.click();
                 return true;
             }, selector, visible);
+
+            // Simulate a true mouse click. The following function scrolls
+            // the element into view, moves the mouse to its center and
+            // presses the left mouse button. This is important for when
+            // an element is above the one we want to click.
+            if (found !== null && typeof found === 'object') {
+                await page.mouse.click(found.x, found.y);
```

Comment:
Wouldn't this click twice if `!visible`?

Reply:
The return value of `page.evaluate` is admittedly overloaded here with multiple meanings:

```
false - Element was not found
null - Element is invisible
true - Clicked on invisible element
{ x: ..., y: ... } - Coordinates for visible element that needs to still be clicked
```